### PR TITLE
Add an option to build with -pthread

### PR DESCRIPTION
**Pull request by abergeron**
I've noticed that the libraries (libtvm.so and friends) are currently linked with pthread, but are not built with support for threading.

I've made this patch to build the libraries with proper thread support.  It is configurable if it needs to be disabled.  It should also work properly on all platforms.

### CODE REVIEW

**Comment by abergeron** [comment]
The test that failed was for batch norm layer.  I think it has nothing to do with my changes, but I'm not 100% sure.

**Comment by tqchen** [comment]
Thanks @abergeron . In the next time when flaky testcase failed, please retrigger the ci by another commit and send a flaky test issue

**Comment by kparzysz-quic** [comment]
What is the intended meaning (and scope) of this flag?  The name of the flag suggests that if `USE_THREADS` if `FALSE`, then the TVM binaries will not use threads at all.  On Hexagon we want to be able to use threads even if cmake's standard scripts cannot detect the proper options, but this flag makes it difficult.

**Comment by abergeron** [comment]
@kparzysz-quic This is mostly for binary compatibility with other C++ libraries.  Building with or without -pthreads affect some of the private C++ classes that end up embedded inside the library and may conflict through the loader.

I have encountered this problem more than once, while loading tvm and pytorch for example.  The python process would crash or hang depending on the load order and gdb indicates incompatible mixed symbols as the most probable cause.

**Comment by kparzysz-quic** [comment]
@abergeron I'm not disputing that adding `-pthread` may be necessary in certain cases.  What I'm asking is what this option is meant to do.  If it's to add `-pthread` to compile/link options, it should be named accordingly, and there should be a way to opt out of it without disabling the use of (p)threads altogether.

**Comment by abergeron** [comment]
Since I'm not touching much, it should still link with pthread as before if you set USE_PTHREADS=OFF.

I'm not sure I would recommend doing that though.
